tracer/src/Datadog.Trace.Coverage.collector/AssemblyProcessor.cs

```diff
@@ -0,0 +1,693 @@
+                                                eHandler.TryEnd == nextInstruction ||
+                                                eHandler.HandlerStart == nextInstruction ||
+                                                eHandler.HandlerEnd == nextInstruction ||
+                                                eHandler.FilterStart == nextInstruction))
+                                        {
+                                            var nextSequencePoint = nextItem.Item2;
+                                            var nextInstructionRange = GetRangeFromSequencePoint(nextSequencePoint);
+                                            var nextInstructionIndex = instructions.IndexOf(nextInstruction);
+
+                                            currentInstruction.OpCode = OpCodes.Ldloca;
+                                            currentInstruction.Operand = coverageScopeVariable;
+                                            instructions.Insert(currentInstructionIndex + 1, Instruction.Create(OpCodes.Ldc_I8, (long)currentInstructionRange));
+                                            instructions.Insert(currentInstructionIndex + 2, Instruction.Create(OpCodes.Ldc_I8, (long)nextInstructionRange));
+                                            instructions.Insert(currentInstructionIndex + 3, Instruction.Create(OpCodes.Call, scopeReport2MethodRef));
+                                            instructions.Insert(currentInstructionIndex + 4, currentInstructionClone);
+
+                                            // We process both instructions in a single call.
+                                            i++;
+                                            continue;
+                                        }
+                                    }
+
+                                    currentInstruction.OpCode = OpCodes.Ldloca;
+                                    currentInstruction.Operand = coverageScopeVariable;
+                                    instructions.Insert(currentInstructionIndex + 1, Instruction.Create(OpCodes.Ldc_I8, (long)currentInstructionRange));
+                                    instructions.Insert(currentInstructionIndex + 2, Instruction.Create(OpCodes.Call, scopeReportMethodRef));
+                                    instructions.Insert(currentInstructionIndex + 3, currentInstructionClone);
+                                }
+
+                                isDirty = true;
+                            }
+                        }
+                    }
+
+                    // Change attributes to drop native bits
+                    if ((module.Attributes & ModuleAttributes.ILLibrary) == ModuleAttributes.ILLibrary)
+                    {
+                        module.Architecture = TargetArchitecture.I386;
+                        module.Attributes &= ~ModuleAttributes.ILLibrary;
+                        module.Attributes |= ModuleAttributes.ILOnly;
+                    }
```

Comment:
The purpose of this block is to remove the **crossgen/ready2run/ngen** (native bits) from the target assembly because we need to ensure our rewritten code will be executed.

So `ModuleAttributes.ILLibrary` indicates that there's native bits in the assembly (https://github.com/jbevain/cecil/pull/475) 

Another thing we have to do is to change the value of `module.Architecture` to `TargetArchitecture.I386` when we have the `ILOnly` option. Because that target architecture indicates native bits architecture. I did several tests and the only way to make it work (and comparing with what Roslyn produces) is to switch to `I386` (There's a couple of Issues and PRs about mixed assemblies in the Mono.Cecil repo, like https://github.com/jbevain/cecil/issues/390)